### ISSUE TITLE
[suggestion] use clazy to check for "inefficient Qt coding mistakes"

### ISSUE THREAD

**Issue by ahjolinna** [cmdrkotori/mpc-qt#80]
I saw the phoronix news about the new tool from KDE team that checks the Qt code quality, I was thinking this would be useful for mpc-qt also, as clazy isn't for KDE usage only it's for all Qt projects...:

> Clazy is a Clang-based static analyzer for Qt that can warn about unnecessary memory allocations, poor API decisions, inefficient usage of some API calls, and more. Additional checks are still being added to Clazy.
> 
> Qt developers wishing to learn more can find out all of the details via this [KDAB blog post](https://www.kdab.com/uncovering-32-qt-best-practices-compile-time-clazy/). The code checker is available via the[ KDE Git repository.](https://github.com/KDE/clazy)

---source [phoronix](http://phoronix.com/scan.php?page=news_item&px=Clang-Clazy-Qt-Slow-Mistakes)

**Comment by cmdrkotori** [comment]
Fixed in 6895f8aa.

**Comment by ahjolinna** [comment]
cool, that was fast, I hope it will be useful in long term...anyway have to try it out and see if I notice any difference

**Comment by mmetak** [comment]
@cmdrkotori Seems that after this commit mpc-qt starts fine, but when opening a file it imediatelly crashes. (archlinux, qt-5.8.0 & qt-5.9.0-beta2)

```
#0  0x00007ffff5c858f5 in ?? () from /usr/lib/libQt5Core.so.5
#1  0x00007ffff5c8bd55 in operator<(QString const&, QString const&) () from /usr/lib/libQt5Core.so.5
#2  0x000000000042f10b in qMapLessThanKey<QString> (key1=..., key2=...) at /usr/include/qt/QtCore/qmap.h:71
#3  0x0000000000434a34 in QMapNode<QString, QVariant>::lowerBound (this=0x13c5e00, akey=...)
    at /usr/include/qt/QtCore/qmap.h:155
#4  0x0000000000431028 in QMapData<QString, QVariant>::findNode (this=0x14024f0, akey=...)
    at /usr/include/qt/QtCore/qmap.h:287
#5  0x000000000042e767 in QMap<QString, QVariant>::operator[] (this=0x7fffffffe050, akey=...)
    at /usr/include/qt/QtCore/qmap.h:664
#6  0x0000000000440ce7 in MpvWidget::self_metadata (this=0x9101c0, metadata=...) at mpvwidget.cpp:674
#7  0x000000000044021d in MpvWidget::ctrl_mpvPropertyChanged (this=0x9101c0, name=..., v=...)
    at mpvwidget.cpp:579
#8  0x000000000044ac89 in QtPrivate::FunctorCall<QtPrivate::IndexesList<0, 1>, QtPrivate::List<QString, QVariant>, void, void (MpvWidget::*)(QString, QVariant)>::call (f=
    (void (MpvWidget::*)(MpvWidget * const, QString, QVariant)) 0x43f8d6 <MpvWidget::ctrl_mpvPropertyChanged(QString, QVariant)>, o=0x9101c0, arg=0x7fffb00252f0) at /usr/include/qt/QtCore/qobjectdefs_impl.h:136
#9  0x000000000044a0f8 in QtPrivate::FunctionPointer<void (MpvWidget::*)(QString, QVariant)>::call<QtPrivate::List<QString, QVariant>, void> (f=
    (void (MpvWidget::*)(MpvWidget * const, QString, QVariant)) 0x43f8d6 <MpvWidget::ctrl_mpvPropertyChanged(QString, QVariant)>, o=0x9101c0, arg=0x7fffb00252f0) at /usr/include/qt/QtCore/qobjectdefs_impl.h:169
#10 0x0000000000448dbe in QtPrivate::QSlotObject<void (MpvWidget::*)(QString, QVariant), QtPrivate::List<QString, QVariant>, void>::impl (which=1, this_=0x910f60, r=0x9101c0, a=0x7fffb00252f0, ret=0x0)
    at /usr/include/qt/QtCore/qobject_impl.h:120
#11 0x00007ffff5e13c79 in QObject::event(QEvent*) () from /usr/lib/libQt5Core.so.5
#12 0x00007ffff7344703 in QWidget::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5
#13 0x00007ffff7302a3c in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /usr/lib/libQt5Widgets.so.5
#14 0x00007ffff730a321 in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQt5Widgets.so.5
#15 0x00007ffff5de6330 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
   from /usr/lib/libQt5Core.so.5
#16 0x00007ffff5de8dfd in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
   from /usr/lib/libQt5Core.so.5
#17 0x00007ffff5e3bd13 in ?? () from /usr/lib/libQt5Core.so.5
#18 0x00007fffeb1317b7 in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
#19 0x00007fffeb131a20 in ?? () from /usr/lib/libglib-2.0.so.0
#20 0x00007fffeb131acc in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0
#21 0x00007ffff5e3c11f in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/libQt5Core.so.5
#22 0x00007ffff5de499a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /usr/lib/libQt5Core.so.5
#23 0x00007ffff5ded284 in QCoreApplication::exec() () from /usr/lib/libQt5Core.so.5
#24 0x00000000004273e2 in Flow::run (this=0x7fffffffe830) at main.cpp:358
#25 0x0000000000422fb3 in main (argc=1, argv=0x7fffffffea58) at main.cpp:39
```

**Comment by ahjolinna** [comment]
I can confirm/replicate this also on openSUSE leap 42.2 using Qt5.8 (repo)

my gdb output: [mpc-qt-log.txt](https://github.com/cmdrkotori/mpc-qt/files/948076/mpc-qt-log.txt)

**Comment by cmdrkotori** [comment]
Unit testing is for the weak.

Crash fixed in 508ddd0.

**Comment by ahjolinna** [comment]
@cmdrkotori : there is only [WAaaGH!!﻿](https://www.youtube.com/watch?v=EQvGuSHMbMc)
